src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls

```diff
@@ -0,0 +1,277 @@
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Settings
+* @group-content ../../ApexDocContent/Settings.htm
+* @description Controller extension for the Data Import Fied Mapping settings panel.
+*/
+public with sharing class STG_PanelDataImportAdvancedMapping_CTRL extends STG_Panel {
+
+    /*******************************************************************************************************
+    * @description is the panel currently Polling for deployment results
+    */
+    public Boolean isPolling {
+        get {
+            if (isPolling == null) {
+                return false;
+            } else {
+                return isPolling;
+            }
+        }
+        private set;
+    }
+
+    /*******************************************************************************************************
+    * @description Maximum timeout for page polling, 12 = 1 minute
+    */
+    private static final Integer MAX_TIME_OUT = 12;
+
+    /*******************************************************************************************************
+    * @description ActionPoller count
+    */
+    private Integer timeoutCount = 0;
+
+    /*******************************************************************************************************
+    * @description Get custom data import settings
+    */
+    public Data_Import_Settings__c dataImportSettings { get; set; }
+
+    /*******************************************************************************************************
+    * @description Boolean used for toggle value in the visualforce page
+    */
+    public Boolean isAdvancedMappingEnabled {
+        get {
+            if (dataImportSettings.Field_Mapping_Method__c == BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING) {
+                return true;
+            }
+            return false;
+        }
+        set;
+    }
+
+    /*******************************************************************************************************
+    * @description is the running user an admin
+    */
+    public Boolean isAdmin {
+        get {
+            return STG_Panel.runningUserIsAdmin();
+        }
+        set;
+    }
+
+    /*******************************************************************************************************
+    * @description Custom metadata deployment result
+    */
+    public Metadata.DeployResult deploymentResult { get; set; }
+
+    /*******************************************************************************************************
+    * @description Boolean that determines if the toggle should be disabled
+    */
+    public Boolean hasDeploymentInProgress { get; set; }
```

Comment:
I think `deploymentResult ` and `hasDeploymentInProgress` should be made private also, since they are only used internally.

Reply:
Don't they need to be public for visualforce page access?

Reply:
Yep, but `deploymentResult`, `migrationMappingUtility` and `dataImportSettings` are all only referenced in the controller and not from the page - so I think they can be private.